src/uk/honeymanga/src/eu/kanade/tachiyomi/extension/uk/honeymanga/HoneyManga.kt

```diff
@@ -37,112 +35,102 @@ class HoneyManga : HttpSource() {
     override val lang = "uk"
     override val supportsLatest = true
 
-    override fun headersBuilder(): Headers.Builder = Headers.Builder()
+    override fun headersBuilder() = super.headersBuilder()
         .add("Origin", baseUrl)
         .add("Referer", baseUrl)
-        .add("User-Agent", userAgent)
 
-    override val client: OkHttpClient = network.client.newBuilder()
+    override val client = network.client.newBuilder()
         .rateLimitHost(API_URL.toHttpUrl(), 10)
         .build()
 
-    // ----- requests -----
+    // ============================== Popular ===============================
+    override fun popularMangaRequest(page: Int) = makeHoneyMangaRequest(page, "likes")
 
-    override fun popularMangaRequest(page: Int): Request {
-        return POST(
-            "$API_URL/v2/manga/cursor-list",
-            headers,
-            makeHoneyMangaRequestBody(page, "likes"),
-        )
-    }
+    override fun popularMangaParse(response: Response) = parseAsMangaResponseDto(response)
 
-    override fun latestUpdatesRequest(page: Int): Request {
-        return POST(
-            "$API_URL/v2/manga/cursor-list",
-            headers,
-            makeHoneyMangaRequestBody(page, "lastUpdated"),
-        )
-    }
+    // =============================== Latest ===============================
+    override fun latestUpdatesRequest(page: Int) = makeHoneyMangaRequest(page, "lastUpdated")
 
+    override fun latestUpdatesParse(response: Response) = parseAsMangaResponseDto(response)
+
+    // =============================== Search ===============================
     override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
         if (query.length >= 3) {
-            return GET(
-                "$SEARCH_API_URL/api/v1/title/search-matching".toHttpUrl()
-                    .newBuilder()
-                    .addQueryParameter("query", query)
-                    .toString(),
-                headers,
-            )
+            val url = "$SEARCH_API_URL/v2/manga/pattern?query=$query"
```

Comment:
Avoid passing strings directly as query parameter as it may cause encoding issues. The previous implementation using the URL builder was more safer.

```suggestion
            val url = "$SEARCH_API_URL/v2/manga/pattern".toHttpUrl().newBuilder()
                .addQueryParameter("query", query)
                .build()
```